src/controllers/hotels.controller.js

```diff
@@ -100,11 +102,70 @@ class Hotel {
       type,
       description,
       cost,
-      status
+      status,
     });
 
     return Responses.handleSuccess(201, 'Room added successfully', res, room);
   }
+
+  /**
+   * @param {object} req
+   * @param {object} res
+   * @returns {object} responses
+   */
+  async getMostVisitedDestination(req, res) {
+    const page = (req.query.page * 1) || 1;
+    const results = [];
+    let trips = [];
+    const requests = await getAllRequestsHotels();
+
+    if (!requests.length) {
+      return Responses.handleError(404, 'There is no past travel yet.', res);
+    }
+
+    const requestTrips = requests.map(el => el.trips);
+
+    for (let i = 0; i < requestTrips.length; i += 1) {
+      trips = trips.concat(requestTrips[i]);
+    }
+
+    const counter = {};
+    for (let i = 0; i < trips.length; i += 1) {
+      const el = trips[i].hotel.id;
+
+      if (counter[el] === undefined) {
+        counter[el] = 1;
+      } else {
+        counter[el] += 1;
+      }
+    }
+
+    const sortable = [];
+
+    // eslint-disable-next-line guard-for-in,no-restricted-syntax
+    for (const element in counter) {
+      sortable.push([element, counter[element]]);
+    }
+    sortable.sort((a, b) => a[1] - b[1]);
+    const five = sortable.slice(page - 1, page * 5);
+    const fiveKeys = five.map(item => item[0]);
+
+    const fivePaginated = await hotelService.mostTravelled(fiveKeys);
```

Comment:
It is hard to know what `five` and `fivePaginated` mean. We can give them better names.

Reply:
I was trying to get the paginated list (5 by 5), but I actually removed that layer as it was done inside the controller.